### ISSUE TITLE
Return Shopify API errors as KeyValuePairs instead of a JSON string

### ISSUE THREAD

**Issue by nozzlegear** [nozzlegear/ShopifySharp#9]
Right now, ShopifySharp returns any errors returned by the Shopify API as a JSON string. The errors are formatted like this:

```
{
    "errors": {
        "addresses.country": [
            "is not a valid country"
        ]
    }
}
```

It would be better if they were returned as a list of KeyValuePairs, which would allow the developer to list them out and sort them by their type.

**Comment by nozzlegear** [comment]
This change is not backwards compatible, and will need to be implemented in version 2.0.

**Comment by nozzlegear** [comment]
**Urgent**: Current implementation will break on [June 1st, 2016](https://docs.shopify.com/api/guides/authentication/oauth#making-authenticated-requests).
